app/models/compute_resources/foreman/model/libvirt.rb

```diff
@@ -124,6 +124,7 @@ def new_vm(attr = { })
     def create_vm(args = { })
       vm = new_vm(args)
       create_volumes :prefix => vm.name, :volumes => vm.volumes, :backing_id => args[:image_id]
+      vm.iso_dir = vm.volumes.first.path.split('/')[0..-2].join('/')
```

Comment:
This assumes that the volume is in a directory storage pool, but it could be something like `/dev/lvmvg/lvmlv`.  I think it probably needs to do two things differently:

1. Check the pool associated with the volume has `type == 'dir'`, and if not, perhaps find another volume that does, or pick the first pool that does(?).
1. Use the path out of the pool rather than parsing it from the volume path.